apps/api/src/donations/donations.service.ts

```diff
@@ -152,14 +154,15 @@ export class DonationsService {
     }
   }
 
-  async getDonationsByUser(personId: string) {
+  async getDonationsByUser(personSub: string) {
+    const person = await this.person.findOneByKeycloakId(personSub)
     const donations = await this.prisma.donation.findMany({
       include: {
         targetVault: {
           include: { campaign: true },
         },
       },
-      where: { personId },
+      where: { personId: person?.id },
     })
```

Comment:
Can you select donation by keycloakId directly without making request for person? Something like where: {person:{keycloakId: personSub}}

Reply:
Done in https://github.com/podkrepi-bg/api/pull/214/commits/b786da485c8a1952a097fd56db318623a7c221d5.
Good suggestion!